app/src/main/java/com/example/cactusnotes/validations/PasswordValidator.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.cactusnotes.validations
+import com.example.cactusnotes.*
+
+class PasswordValidator : Validator {
+
+    override fun validate(field: String) = when {
+        field.isEmpty() -> { ValidationResult(false, R.string.password_empty) }
+        field.length < 7 -> { ValidationResult(false, R.string.password_short) }
+        field.length > 40 -> { ValidationResult(false, R.string.password_long) }
+        !field.containsRequiredCharCharacters() -> { ValidationResult(false, R.string.password_special_character) }
+        else -> { ValidationResult(true, null) }
+    }
+
+    private fun String.hasDigits() = any { it.isDigit() }
+
+    private fun String.hasUpperCaseLetters() = any { it.isUpperCase() }
+
+    private fun String.hasLowerCaseLetters() = any { it.isLowerCase() }
+
+    private fun String.hasSpecialLetters() = any { !it.isLetterOrDigit() }
+
+    private fun String.containsRequiredCharCharacters() = hasDigits() && hasLowerCaseLetters()
```

Comment:
typo in function name:

❌ containsRequiredCharCharacters
✅ containsRequiredCharacters